cmd/monaco/dynatrace/dynatrace.go

```diff
@@ -33,92 +30,87 @@ import (
-	if !featureflags.VerifyEnvironmentType.Enabled() {
-		return true
-	}
+var ErrorMissingAuth = errors.New("no authentication credentials provided")
+
+// VerifyEnvironmentsAuthentication verifies that all environments can be reached with the defined credentials.
+// It returns the first error encountered
+func VerifyEnvironmentsAuthentication(ctx context.Context, envs manifest.EnvironmentDefinitionsByName) error {
 	for _, env := range envs {
-		if !isValidEnvironment(ctx, env) {
-			return false
+		if err := VerifyEnvironmentAuthentication(ctx, env); err != nil {
+			return err
 		}
 	}
-	return true
+	return nil
 }
 
-func isValidEnvironment(ctx context.Context, env manifest.EnvironmentDefinition) bool {
+// VerifyEnvironmentAuthentication checks if the provided token and OAuth credentials of the provided environment are valid.
+func VerifyEnvironmentAuthentication(ctx context.Context, env manifest.EnvironmentDefinition) error {
 	if env.Auth.Token == nil && env.Auth.OAuth == nil {
-		report.GetReporterFromContextOrDiscard(ctx).ReportLoading(report.StateError, errors.New("no token or oAuth credentials provided in the manifest"), "", nil)
-		log.Error("No token or oAuth credentials provided in the manifest")
-		return false
+		return ErrorMissingAuth
 	}
 
-	if env.Auth.OAuth == nil {
-		return canEstablishClassicConnection(ctx, env)
+	classicUrl := env.URL.Value
+
+	// check if the OAuth connection works and get the classicURL in order to check the token authentication next if given
+	if env.Auth.OAuth != nil {
+		oauthCreds := clientcredentials.Config{
+			ClientID:     env.Auth.OAuth.ClientID.Value.Value(),
+			ClientSecret: env.Auth.OAuth.ClientSecret.Value.Value(),
+			TokenURL:     env.Auth.OAuth.GetTokenEndpointValue(),
+		}
+		var err error
+		if classicUrl, err = getDynatraceClassicURL(ctx, env.URL.Value, oauthCreds); err != nil {
+			err = formatAuthError(env, env.URL.Value, err, "OAuth")
+			return fmt.Errorf("please verify that this environment is a Dynatrace Platform environment. %w", err)
+		}
 	}
 
-	return isPlatformEnvironment(ctx, env)
+	if env.Auth.Token != nil {
+		if err := validateEstablishClassicConnection(ctx, env, classicUrl); err != nil {
+			return err
+		}
+	}
+	return nil
 }
 
-// canEstablishClassicConnection checks if a classic connection (via token) can be established. Scopes are not validated.
-func canEstablishClassicConnection(ctx context.Context, env manifest.EnvironmentDefinition) bool {
+// validateEstablishClassicConnection checks if a classic connection (via token) can be established. Scopes are not validated.
+func validateEstablishClassicConnection(ctx context.Context, env manifest.EnvironmentDefinition, classicURL string) error {
 	token := env.Auth.Token.Value.Value()
 	client, err := clients.Factory().
-		WithClassicURL(env.URL.Value).
+		WithClassicURL(classicURL).
 		WithAccessToken(token).
 		WithRateLimiter(true).
 		WithRetryOptions(&client.DefaultRetryOptions).
 		CreateClassicClient()
```

Comment:
Do we also have to explicitly enable the support archive here too?

Reply:
Good catch! Forgot about this one.
Added 👍